app/src/main/java/org/systers/mentorship/view/activities/LoginActivity.kt

```diff
@@ -46,8 +59,46 @@ class LoginActivity : BaseActivity() {
                     startActivity(intent)
                     finish()
                 } else {
-                    Snackbar.make(getRootView(), loginViewModel.message, Snackbar.LENGTH_LONG)
+                    if(loginViewModel.verified)
+                        Snackbar.make(getRootView(), loginViewModel.message, Snackbar.LENGTH_LONG)
                             .show()
+                    else
+                        Snackbar.make(getRootView(), R.string.not_verified, Snackbar.LENGTH_INDEFINITE)
+                                .setAction(R.string.resend_email) {
+                                    val data = Email(username)
+                                            userDataManager.resendEmail(data)
+                                            .subscribeOn(Schedulers.newThread())
+                                            .observeOn(AndroidSchedulers.mainThread())
+                                            .subscribe(object: DisposableObserver<CustomResponse>() {
+                                                override fun onNext(customResponse: CustomResponse) {
+                                                    message = customResponse.message
+                                                    Snackbar.make(getRootView(), message, Snackbar.LENGTH_SHORT).show()
+                                                }
```

Comment:
I don't think snackbar is the best place to prompt the user to resend the email.